Add Two-Factor Authentication (2FA) functionality

ADG08 opened the pull request:
- Implemented 2FA setup, verification, and management in user model and controllers.
- Updated login and logout processes to handle 2FA.
- Created a new TwofaController for handling 2FA-related requests.
- Added frontend components for 2FA settings and verification.
- Updated routes to include 2FA endpoints.
- Enhanced user model with necessary fields for 2FA.
- Added validation for 2FA codes.
- Updated language files for 2FA messages.

ADG08 committed:
Add Two-Factor Authentication (2FA) functionality

- Implemented 2FA setup, verification, and management in user model and controllers.
- Updated login and logout processes to handle 2FA.
- Created a new TwofaController for handling 2FA-related requests.
- Added frontend components for 2FA settings and verification.
- Updated routes to include 2FA endpoints.
- Enhanced user model with necessary fields for 2FA.
- Added validation for 2FA codes.
- Updated language files for 2FA messages.

Loeryx commented on inertia/components/profile/TwoFactorSettings.vue:
Met l'i18n et met le script au début du fichier please

Loeryx commented on inertia/pages/auth/login.vue:
```diff
@@ -55,91 +87,173 @@ const { t } = useI18n()
-              <button
-                type="submit"
-                :disabled="form.processing"
-                class="bg-[#5C4741] text-white px-4 py-2 rounded-lg font-altone text-sm md:text-base flex-1 cursor-pointer"
```
Il fonctionne le disabled ?

Loeryx commented on inertia/pages/auth/login.vue:
```diff
@@ -55,91 +87,173 @@ const { t } = useI18n()
+          <div v-else class="w-full md:w-1/2 flex flex-col items-center justify-center p-6 md:p-10">
+            <form @submit.prevent="twofaForm.post('/login/verify-2fa')" class="w-full">
+              <div class="text-center mb-6">
+                <h2 class="text-xl font-altone mb-2">Two-Factor Authentication</h2>
```
i18n

ADG08 committed:
Enhance Two-Factor Authentication (2FA) user interface and functionality

- Refactored TwoFactorSettings component to utilize Vue's Composition API.
- Added internationalization support for 2FA messages in English and French.
- Improved user experience with dynamic loading states and error handling during 2FA setup and verification.
- Updated login page to include 2FA prompts and messages.
- Enhanced language files with additional 2FA-related strings for better user guidance.

ADG08 committed:
Merge branch 'dev' into PA-77

ADG08 committed:
Update login page and language files for consistency and clarity

- Updated the login page to ensure the "Forgot Password" link points correctly to the intended route.
- Added missing commas in English and French language files for proper JSON formatting.
- Enhanced error message consistency in profile language files.
